### PR TITLE
add new component input array

### PR DESCRIPTION
**Pull request by WaQasRPL**
https://github.com/boxture/oms/issues/2440

### CODE REVIEW

File: app/components/satis/input_array/component.html.slim [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    template data-satis-input-array-target="rowTemplate"
+        .flex data-satis-input-array-target="inputRow"
+            div.flex-auto
+                = text_field_tag  nil, nil, class: 'form-control mt-1', data: { action: "input->satis-input-array#input" }
+            div.flex-none.shrink-0
+                button.w-10.h-full.items-center data-action="click->satis-input-array#removeRow"
+                    i.fas.fa-xmark
+
+    div data-satis-input-array-target="rowsContainer"
+        - if options[:sub_attribute].present?
```
Comment (tdegrunt) [inline app/components/satis/input_array/component.html.slim:22]: For sub_attribute, is the intent to indicate what the value should be, or what the value attribute is? Perhaps we should call it value_method in line with this:

https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/actionview/lib/action_view/helpers/form_options_helper.rb#L400

File: app/components/satis/input_array/component.html.slim [hunk 2]
```diff
@@ -0,0 +1,41 @@
+div.satis-input-array data-controller="satis-input-array" data-satis-input-array-input-name-value="#{options[:sub_attribute].present? ? "#{form.object_name}[#{attribute.to_s}][][#{options[:sub_attribute]}]" : "#{form.object_name}[#{attribute.to_s}][]"}"
+    / loops incorrect number of time, some unknown issue
```
Comment (tdegrunt) [inline app/components/satis/input_array/component.html.slim:2]: Remove comments

File: app/components/satis/input_array/component.html.slim [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        - else
+            - options[:collection].each do |field|
+                .flex data-satis-input-array-target="inputRow"
+                    div.flex-auto
+                        = text_field_tag  "#{form.object_name}[#{attribute.to_s}][]", field, class: 'form-control mt-1'
+                    div.flex-none.shrink-0
+                        button.w-10.h-full.items-center data-action="click->satis-input-array#removeRow"
+                            i.fas.fa-xmark
+    div.mt-3
+        button.button.secondary data-action="click->satis-input-array#addRow"
```
Comment (tdegrunt) [inline app/components/satis/input_array/component.html.slim:40]: Should Add Row just not show a + icon?

**Review by tdegrunt — CHANGES_REQUESTED** [review]

File: app/components/satis/input_array/component.html.slim (hunk 2 above)
Comment (WaQasRPL) [reply app/components/satis/input_array/component.html.slim]: Done

File: app/components/satis/input_array/component.html.slim (hunk 3 above)
Comment (WaQasRPL) [reply app/components/satis/input_array/component.html.slim]: Updated

File: app/components/satis/input_array/component.html.slim (hunk 1 above)
Comment (WaQasRPL) [reply app/components/satis/input_array/component.html.slim]: sub_attribute renamed to value_method

**Comment by tdegrunt** [comment]
Does this also have the logic in place to add a row when you change/fill the field?

**Comment by WaQasRPL** [comment]
tabindex removed from add and delete button

**Pull request merged**